aptos-move/framework/aptos-stdlib/sources/cryptography/bls12381.move

```diff
@@ -226,6 +266,90 @@ module aptos_std::bls12381 {
         verify_signature_share_internal(signature_share.bytes, public_key.bytes, message)
     }
 
+    fun encode_items<T: copy+drop>(items: &vector<T>): vector<vector<u8>> {
+        let item_count = std::vector::length(items);
+        let messages = vector[];
+        let i = 0;
+        while (i < item_count) {
+            let item = std::vector::borrow(items, i);
+            let encoded = new_signed_message(*item);
+            std::vector::push_back(&mut messages, bcs::to_bytes(&encoded));
+            i = i + 1;
+        };
+        messages
+    }
+
+    #[test_only]
+    public fun generate_keys(): (SecretKey, PublicKeyWithPoP) {
+        let (sk_bytes, pk_bytes) = generate_keys_internal();
+        let sk = SecretKey {
+            bytes: sk_bytes
+        };
+        let pkpop = PublicKeyWithPoP {
+            bytes: pk_bytes
+        };
+        (sk, pkpop)
+    }
+
+    #[test_only]
+    public fun sign_arbitrary_bytes(signing_key: &SecretKey, message: vector<u8>): Signature {
+        Signature {
+            bytes: sign_internal(signing_key.bytes, message)
+        }
+    }
+
+    #[test_only]
+    public fun sign_struct<T: drop>(signing_key: &SecretKey, data: T) : Signature {
+        let encoded = ed25519::new_signed_message(data);
+        Signature {
+            bytes: sign_internal(signing_key.bytes, bcs::to_bytes(&encoded))
+        }
+    }
+
+    #[test_only]
+    public fun multi_sign_arbitrary_bytes(signing_keys: &vector<SecretKey>, messages: vector<u8>): AggrOrMultiSignature {
+        let n = std::vector::length(signing_keys);
+        let sigs = vector[];
+        let i: u64 = 0;
+        while (i < n) {
+            let sig = sign_arbitrary_bytes(std::vector::borrow(signing_keys, i), messages);
+            std::vector::push_back(&mut sigs, sig);
+            i = i + 1;
+        };
+        let multisig = aggregate_signatures(sigs);
+        option::extract(&mut multisig)
+    }
+
+    #[test_only]
+    public fun multi_sign_struct<T: drop>(signing_keys: &vector<SecretKey>, data: T) : AggrOrMultiSignature {
```

Comment:
same here

Reply:
By reference is better.

Reply:
But let's not add this at all, since it uses the old domain separation.